application/src/Form/SiteSettingsForm.php

```diff
@@ -124,6 +134,32 @@ public function init()
             ],
         ]);
 
+        $langFieldset->add([
+            'name' => 'filter_locale_values',
+            'type' => 'checkbox',
+            'options' => [
+                'label' => 'Filter values based on site localization', // @translate
+                'info' => 'Check to show only values matching the site language setting and values without locale ID.',
+            ],
+            'attributes' => [
+                'id' => 'filter_locale_values',
+                'value' => (bool) $settings->get('filter_locale_values', false),
+            ],
+        ]);
+
+        $langFieldset->add([
```

Comment:
This feature should _also_ probably be its own thing (for one it's much simpler and therefore easier to merge) but I think we can probably deal with them together.

We could also merge it separately, though.

Reply:
Once you filter the values based on the language, you end up with item show pages cluttered with now kind of useless labels, a lot of them. That's my experience on this but if your really prefer to separate this, please tell me.

Reply:
I think It'll be better as a separate PR, though I do agree it becomes more important to have with the "filter by locale" feature turned on.

Among other reasons, It'll be easier for us to track it as a separate new feature in release notes and things like that.

A side note: even though it's only checked once, I think the same "cast to bool" advice applies with this check as well.

Reply:
OK removed in 1acfc9f